### PR TITLE
Feature/unit testing

### PR DESCRIPTION
**Pull request by OvidiuGi**
Still have to review the code to be PSR compliant and maybe add more cases to the tests.
Lmk if you find something to modify :)

### CODE REVIEW

File: tests/Validator/CnpValidatorTest.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            $this->assertNoViolation();
+        } else {
+            $this->buildViolation('This is not a valid CNP')->assertRaised();
+        }
+    }
+
+    public function provideCnpValues(): array
+    {
+        return [
+            ['5010911070069', true, ['test']],
```
Comment (marceldobrica) [inline tests/Validator/CnpValidatorTest.php:33]: It is strange.... testCnp requires 2 arguments.... 3 provided by privideCnpValues...

File: tests/Validator/PasswordValidatorTest.php [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    public function providePasswordValues(): array
+    {
+        return[
+            ['', false, ['test']],
+            ['parola', false, ['test']],
+            ['parola!!!!!', false, ['test']],
+            ['Parola11!@', true, ['test']],
+            ['Parol a12!', false, ['test']],
+        ];
+    }
```
Comment (marceldobrica) [inline tests/Validator/PasswordValidatorTest.php:39]: Same as above ... providePassowrdValues should return an array without last element ['test'].

File: tests/Validator/PasswordValidatorTest.php [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+    protected function createValidator(): PasswordValidator
+    {
+        return new PasswordValidator();
+    }
+}
```
Comment (marceldobrica) [inline tests/Validator/PasswordValidatorTest.php:45]: PSR-12 - new line at and of each file

File: tests/Controller/ArgumentValueResolver/UserDtoArgumentValueResolverTest.php [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+        $userDto->firstName = 'Ovidiu';
+        $userDto->lastName = 'Gireada';
+        $userDto->email = 'example@chiwawa.wawa';
+        $userDto->password = 'MySecretPassword';
+        $userDto->confirmPassword = 'MySecretPassword';
+        $userDto->cnp = '5010911070069';
+
+        self::assertEquals($userDto,$dto);
+    }
+}
```
Comment (marceldobrica) [inline tests/Controller/ArgumentValueResolver/UserDtoArgumentValueResolverTest.php:66]: PSR-12

File: src/Validator/CnpValidator.php [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    {
+        if (!$constraint instanceof Cnp) {
+            throw new UnexpectedTypeException($constraint, Cnp::class);
+        }
+
+        $regexResponse = preg_match_all(
+            '/^([1-8])(\d{2})(0[1-9]|1[0-2])(0[1-9]|[1-2]\d|3[0-1])(0\d|[1-3]\d|4[0-8])(\d{3})(\d)$/',
+            $value,
+            $matches
+        );
```
Comment (marceldobrica) [inline src/Validator/CnpValidator.php:21]: Counties 51 (Călărași) and 52(Giurgiu) are not in your regex expression... https://ro.wikipedia.org/wiki/Cod_numeric_personal_(Rom%C3%A2nia).... probably you have looked on another cnp description?
--


[Călărași](https://ro.wikipedia.org/wiki/Jude%C8%9Bul_C%C4%83l%C4%83ra%C8%99i) and 52

File: tests/Validator/PasswordValidatorTest.php (hunk 3 above)
Comment (OvidiuGi) [reply tests/Validator/PasswordValidatorTest.php:45]: I did, but i guess github deleted the new line..

File: tests/Controller/ArgumentValueResolver/UserDtoArgumentValueResolverTest.php (hunk 4 above)
Comment (OvidiuGi) [reply tests/Controller/ArgumentValueResolver/UserDtoArgumentValueResolverTest.php]: fixed :)

File: tests/Validator/PasswordValidatorTest.php (hunk 2 above)
Comment (OvidiuGi) [reply tests/Validator/PasswordValidatorTest.php:39]: It is something optional, it works with it.

**Pull request merged**